### PR TITLE
[7.2] Validate stack modules' metricsets when xpack.enabled = true (#12386)

### PR DESCRIPTION
**Pull request by ycombinator**
Backports the following commits to 7.2:
 - Validate stack modules' metricsets when xpack.enabled = true  (#12386)

### CODE REVIEW

File: CHANGELOG.next.asciidoc [hunk 1]
```diff
@@ -218,6 +218,10 @@ https://github.com/elastic/beats/compare/v7.0.0-alpha2...master[Check the HEAD d
 - Add AWS cloudwatch metricset. {pull}11798[11798] {issue}11734[11734]
 - Add `regions` in aws module config to specify target regions for querying cloudwatch metrics. {issue}11932[11932] {pull}11956[11956]
 - Keep `etcd` followers members from reporting `leader` metricset events {pull}12004[12004]
+- Add overview dashboard to Consul module {pull}10665[10665]
```
Comment (cachedout) [inline CHANGELOG.next.asciidoc]: A few of these entries look unrelated and are [not contained in the original PR](https://github.com/elastic/beats/pull/12386/files#diff-8a3db71464dcd8830047f8fb92084b27R242). Did these get included accidentally.

Comment (ycombinator) [reply CHANGELOG.next.asciidoc]: Yes, they were indeed 😞. Thanks for catching this. Here's the PR that fixes up the CHANGELOG in `7.2` again: https://github.com/elastic/beats/pull/12474.
